00xx-account-deletion.md

```diff
@@ -0,0 +1,384 @@
+  - Validate the absence of impact of the feature on the existing functionalities.
+- **Testing approach**:
+  - Unit tests for the ledger implementations;
+  - Integration tests for the extension of transaction logics.
+- **Testing scope**:
+  - Testing sequences of intertwined addition/removal commands;
+  - Testing simple high-level account updates;
+  - Testing from `o1js` typical use-cases scenarios.
+- **Testing requirements**:
+  - Test in various local setups to simulate different development environments.
+- **Testing resources**:
+  - Simulated environments resembling typical local development setups.
+
+
+## Drawbacks
+
+The main drawback of implementing account deletion is that it needs a hard fork
+to be activated on-chain.
+
+On the surface, the changes might seem trivial (adding a removal function,
+adding a field in a data structure) but they they actually have a widespread
+impact on the codebase (50+ files are touched just for adding the aforementioned
+field addition so that the our codebase compiles). We thus need to be extra
+careful both in the coding *and* the reviewing phases.
+
+## <a name="alternatives"></a>Rationale and alternatives
+
+### Merkle trees
+
+For the sake of completeness, let us consider another option for implementing
+removal at the level of Merkle trees, which aims at maintaining the fact that
+any new insertions happens on the leftmost available location, and that all
+leaves between the leftmost one and the rightmost non-empty one are filled.
+
+In this case, removing `C` would not update any free list -- since this solution
+does not need this concept. But it would change the location of `D`.
+
+```
+    H(H(A,B),H(D,x))
+      /       \
+   H(A,B)     H(D,x)
+   /  \       /   \
+  A    B     D     x
+```
+
+Since the ledger implementation also tracks mapping from account to location as
+well as location to accounts, both these mappings would need updating.
+
+Now adding data `E` would result in
+
+```
+    H(H(A,B),H(D,E))
+      /       \
+   H(A,B)     H(D,E)
+   /  \       /  \
+  A    B     D    E
+```
+
+##### Implementation choice
+
+We opt to implement the tracking of freed locations as illustrate in [this section](#merkle_trees).
+
+While this choice implies adding a structure in the masking ledger
+implementation to track freed locations, it offers 2 advantages:
+
+* the cost of removal is slightly better since we do not need to update location
+   <-> account mappings;
+*  account locations in this representation are fixed for the lifetime of the
+  account: proof of inclusion would not change but for a single element of the
+  Merkle path.
+
+
+## Prior art
+
+## Unresolved questions
+
+- Merkelization of ledger + free list and interface for interacting with this data structure
+
+
+## Resources - TB removed
+
+
+> Here are some things I am thinking about, maybe they help you inform the RFC:
+>
+> - easy API - we need to express account deletion via an easy API in the sdk
```

Comment:
The SDK should only be interfacing with the `Account_update.t` portion, so the API is 'handling and extra boolean' at that level, plus some wrapping TBD.

Reply:
This part should be commented out. This was just for me to check that these points are addressed in the RFC